### PR TITLE
feat: `generateSignInUrl()`

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

implement `generateSignInUrl()` according to SDK convention

## Testing

UT

## Reviwers

@logto-io/eng

### CODE REVIEW

File: Sources/Logto/Core/LogtoCore.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        }
+
+        let baseQueryItems = [
+            URLQueryItem(name: "client_id", value: clientId),
+            URLQueryItem(name: "redirect_uri", value: redirectUri.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)),
+            URLQueryItem(name: "code_challenge", value: codeChallenge),
+            URLQueryItem(name: "code_challenge_method", value: LogtoCore.codeChallengeMethod),
+            URLQueryItem(name: "state", value: state),
+            URLQueryItem(name: "scope", value:
+                (["openid", "offline_access"] + (scope?.inArray ?? [])).joined(separator: " ")),
```
Comment (simeng-li) [inline Sources/Logto/Core/LogtoCore.swift:39]: just want to align our usage of r default OIDC Provide scopes, following the SDK convention doc:
1. The is `scope` param defined in the `LogtoConfig` object for all platform SDKs. Which mentioned, `['openid', 'offline_access']` are always included. 
2. Here at the signinUrl generate methods or Core SDK also have a similar approach.  
should we keep two source of there scopes? cc: @xiaoyijun

**Review by simeng-li — APPROVED** [review]
overall LGTM

File: Sources/Logto/Core/LogtoCore.swift (hunk 1 above)
Comment (gao-sun) [reply Sources/Logto/Core/LogtoCore.swift:39]: good call. i would prefer to remove mandatory `['openid', 'offline_access']` here and keep the logic in `LogtoConfig` init process.

Comment (gao-sun) [reply Sources/Logto/Core/LogtoCore.swift]: opened thread in slack

Comment (xiaoyijun) [reply Sources/Logto/Core/LogtoCore.swift]: I prefer to add the insurance logic code in both `core` and `LogtoCongifg`, for users may use them independently.

**Pull request merged**